h/api/models/document.py

```diff
@@ -26,26 +26,40 @@ class Document(Base, mixins.Timestamps):
     def __repr__(self):
         return '<Document %s>' % self.id
 
+    @property
+    def title(self):
+        titles = [m.value for m in self.meta if m.type == 'title']
+
+        if titles:
+            return titles[0]
+
     @classmethod
-    def find_or_create_by_uris(cls, claimant_uri, uris,
+    def find_by_uris(cls, session, uris):
+        """
+        Find documents by a list of uris.
+        """
+        query_uris = [text_type(uri.normalize(u), 'utf-8') for u in uris]
+
+        matching_claims = (session.query(DocumentURI).filter(
+                DocumentURI.uri_normalized.in_(query_uris))
+                .distinct(DocumentURI.document_id).subquery())
+        matching_claims_aliased = aliased(DocumentURI, matching_claims)
+
+        return session.query(Document).join(matching_claims_aliased)
+
+    @classmethod
+    def find_or_create_by_uris(cls, session, claimant_uri, uris,
                                created=None, updated=None):
         """
-        Find or create documents from a claimant uri and a set of uris.
+        Find or create documents from a claimant uri and a list of uris.
 
         It tries to find a document based on he claimant and the set of uris.
         If none can be found it will return a new document with the claimant
         uri as its only document uri as a self-claim.
```

Comment:
It might be worth explicitly noting here that it is up to the caller to create appropriate DocumentURI claims for the *other* URIs (`uris`) if necessary.